sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -164,11 +164,42 @@ case class Join(
         left.output.map(_.withNullability(true)) ++ right.output
       case FullOuter =>
         left.output.map(_.withNullability(true)) ++ right.output.map(_.withNullability(true))
+      case NaturalJoin(jt) =>
+        outerProjectList(jt).map(_.toAttribute)
       case _ =>
         left.output ++ right.output
     }
   }
 
+  def outerProjectList(jt: JoinType): Seq[NamedExpression] = {
```

Comment:
also i'm not 100% sure, but i suspect if we make resolve false when the plan is natural join type, then we can just move this function into the analyzer. it really doesn't belong in the join node itself.